src/EditorFeatures/VisualBasicTest/UseSystemHashCode/UseSimpleUsingStatementTests.vb

```diff
@@ -0,0 +1,627 @@
+
+class B
+    public overrides function GetHashCode() as integer
+        Return 0
+    end function
+end class
+
+class C
+    inherits B
+    dim j as integer
+
+    public overrides function $$GetHashCode() as integer
+        dim hashCode = 339610899
+        hashCode = hashCode * -1521134295 + MyBase.GetHashCode()
+        Return hashCode
+    end function
+end class",
+" namespace System
+    public structure HashCode
+    end structure
+end namespace
+
+class B
+    public overrides function GetHashCode() as integer
+        Return 0
+    end function
+end class
+
+class C
+    inherits B
+    dim j as integer
+
+    public overrides function GetHashCode() as integer
+        Return System.HashCode.Combine(MyBase.GetHashCode())
+    end function
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseSystemHashCode)>
+        Public Async Function TestFieldAndProp() As Task
+            Await TestInRegularAndScriptAsync(
+"imports System.Collections.Generic
+namespace System
+    public structure HashCode
+    end structure
+end namespace
+
+class C 
+    dim i as integer
+
+    readonly property S as string
+
+    public overrides function $$GetHashCode() as integer
+        dim hashCode = -538000506
+        hashCode = hashCode * -1521134295 + i.GetHashCode()
+        hashCode = hashCode * -1521134295 + EqualityComparer(of string).Default.GetHashCode(S)
+        Return hashCode
+    end function
+end class",
+"imports System.Collections.Generic
+namespace System
+    public structure HashCode
+    end structure
+end namespace
+
+class C 
+    dim i as integer
+
+    readonly property S as string
+
+    public overrides function GetHashCode() as integer
+        Return System.HashCode.Combine(i, S)
+    end function
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseSystemHashCode)>
+        Public Async Function TestNotOnNonGetHashCode() As Task
+            Await TestMissingAsync(
+"imports System.Collections.Generic
+namespace System
+    public structure HashCode
+    end structure
+end namespace
+
+class C 
+    dim i as integer
+
+    readonly property S as string
+
+    public overrides function $$GetHashCode1() as integer
+        dim hashCode = -538000506
+        hashCode = hashCode * -1521134295 + i.GetHashCode()
+        hashCode = hashCode * -1521134295 + EqualityComparer(of string).Default.GetHashCode(S)
+        Return hashCode
+    end function
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseSystemHashCode)>
+        Public Async Function TestNotWithoutReturn() As Task
+            Await TestMissingAsync(
+"imports System.Collections.Generic
+namespace System
+    public structure HashCode
+    end structure
+end namespace
+
+class C 
+    dim i as integer
+
+    readonly property S as string
+
+    public overrides function $$GetHashCode() as integer
```

Comment:
Not sure if you want a test where you're using the implicit variable that matches the function name in VB. Even if people aren't doing it, I could imagine the refactoring being confused by it somehow...

Reply:
To clarify: I'm OK with not actually offering the fix in the VB implicit function return case, I just don't want to crash on it.

Reply:
Great idea!  test added!